### PR TITLE
DData shard store improvements

### PR DESCRIPTION
**Pull request by johanandren**
* Play more nicely with the load timeout
* Load all entities up front

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala [hunk 1]
```diff
@@ -86,8 +88,8 @@ private[akka] final class DDataRememberEntitiesShardStore(
   implicit val selfUniqueAddress: SelfUniqueAddress = SelfUniqueAddress(node.selfUniqueAddress)
 
   private val readMajority = ReadMajority(settings.tuningParameters.waitingForStateTimeout, majorityMinCap)
-  // Note that the timeout is actually updatingStateTimeout x 3 since we do 3 retries
-  private val writeMajority = WriteMajority(settings.tuningParameters.updatingStateTimeout, majorityMinCap)
+  // Note that the timeout is actually updatingStateTimeout / 4 so that we fit 3 retries and a response in the timeout before the shard sees it as a failure
+  private val writeMajority = WriteMajority(settings.tuningParameters.updatingStateTimeout / 4, majorityMinCap)
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala:92]: @chbatey not 100% sure this is a good solution, have any better idea ?

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala (hunk 1 above)
Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/DDataRememberEntitiesShardStore.scala:92]: I think this is fine. No harm in parts of the get all timing out and retrying

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Hmm, same lease tests failing again (I thought that was unrelated)

**Pull request merged**